tests/integration/devfile/cmd_devfile_push_test.go

```diff
@@ -88,6 +88,21 @@ var _ = Describe("odo devfile push command tests", func() {
 			helper.CmdShouldPass("odo", "push", "--devfile", "devfile.yaml", "--project", namespace)
 		})
 
+		It("checks that odo push works outside of the context directory", func() {
+			helper.CmdShouldPass("git", "clone", "https://github.com/che-samples/web-nodejs-sample.git", projectDirPath)
```

Comment:
I left the `git clone` in since https://github.com/openshift/odo/pull/3065 isn't merged yet.

If https://github.com/openshift/odo/pull/3065 goes in before this PR is merged, I'll update the PR to copy the project instead of cloning it.